### ISSUE TITLE
Piano Roll: adding button "Quick clone" (copying to the next beat)

### ISSUE THREAD

**Issue by vlad0337187** [LMMS/lmms#3015]
Hello dear friends.

I'm very grateful you for that I can use such good DAW without copyright infringement or paying big money for this.

As very often I need to copy some highlighted area from one beat to the next one,
so I suggest to add such button, that will:
1. copy and paste notes from current beat to the next one (current - beat, in which the last touched (created, or just clicked) note is. (if no notes are highlighted);
2. copy and paste highlighted notes from current beat\beats to the next ones (if some notes are highlighted).

The button will appear in the last pattern, in which:
1. highlighted notes are present (if some notes are highlighted);
2. the last touched note is present (if no notes are highlighted).

Example:
![button](https://cloud.githubusercontent.com/assets/12682937/18232414/c2e164a6-72d7-11e6-89e6-25afbee99322.png)
![button_2](https://cloud.githubusercontent.com/assets/12682937/18232413/c2dfd7c6-72d7-11e6-8f9c-7e906c184100.png)

Best regards,
Vladislav.

**Comment by Spekular** [comment]
I'd say it's fairly easy to do this already with Shift-click-drag. Is that too slow or is there another issue?

**Comment by vlad1777d** [comment]
Thank you for answer. Yes, I can do so. But this way means lesser amount of clicks. 
I need to press "highlight notes" button, than shift-click-drag, than i need not to be mistaken while I drag notes.

And what about if I (or you) have many beats with notes. And if you highlighted them, and pressed shift-click:
    -on the last note: you need to drag all notes through all beats to the end, this means more actions. And if the first note cannot be seen, it's behind the edge of screen, you need dragging several times.
    -on the first note: the same, you need to drag them through all beats.

There are many similar situations, this only enhances the usability of program.

**Comment by tresf** [comment]
> -on the last note: you need to drag all notes through all beats to the end, this means more actions.

The way to do it is to copy the notes and then move the play marker and then paste, which is an explicit action.  The above proposal on the other hand is an implicit action, it makes assumptions about what the user wants to do.

We have this `clone` concept available in the Beat/Bassline Editor but that is a bit more controlled as the patterns have a fixed length.

Can you please update the mockup to include a before/after illustration?

I would recommend this feature is called `Quick clone` and I'd also recommend that when the button is hovered-over it auto-hightlights the notes that will be cloned as well as makes a ghost-preview of what is going to be drawn so the user knows exactly what is going to happen when it is clicked.

Another potential problem with this implementation is the UI already uses that area for drawing the loop points, so it will ultimately have a cluttering situation going on when they overlap.

**Comment by vlad1777d** [comment]
Yes, I can. Do you mean to include my general screenshots without editing?

Yes, I did not thought about the problem, you described in last paragraph.

**Comment by tresf** [comment]
> Do you mean to include my general screenshots without editing?

No, sorry, something showing the before/after of the cloned patterns.  e.g. https://github.com/LMMS/lmms/pull/1844#issuecomment-77930706

**Comment by vlad1777d** [comment]
Ok, thank you, I'll edit this post with accounting of your recommendations and I'll think about possible errors and solving them.

**Comment by tresf** [comment]
As part of a pruning effort, this `enhancement` request is archived into a dedicated "Better Workflow" checklist here #4877.
